src/main/java/edu/ucsb/cs156/example/controllers/RecommendationController.java

```diff
@@ -0,0 +1,120 @@
+import edu.ucsb.cs156.example.repositories.RecommendationRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+import java.time.LocalDateTime;
+
+@Api(description = "Recommendations")
+@RequestMapping("/api/Recommendation")
+@RestController
+@Slf4j
+public class RecommendationController extends ApiController {
+
+    @Autowired
+    RecommendationRepository recommendationRepository;
+
+    @ApiOperation(value = "List all recommendations")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<Recommendation> allRecommendations() {
+        Iterable<Recommendation> recommendations = recommendationRepository.findAll();
+        return recommendations;
+    }
+
+    @ApiOperation(value = "Get a single recommendation")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public Recommendation getById(
+            @ApiParam("id") @RequestParam Long id) {
+        Recommendation recommendation = recommendationRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(Recommendation.class, id));
+
+        return recommendation;
+    }
+
+    @ApiOperation(value = "Create a new recommendation")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/post")
+    public Recommendation postRecommendation(
+            @ApiParam("requesterEmail") @RequestParam String requesterEmail,
+            @ApiParam("professorEmail") @RequestParam String professorEmail,
+            @ApiParam("explanation") @RequestParam String explanation,
+            @ApiParam("date (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("dateRequested") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime dateRequested,
+            @ApiParam("date needed by (in iso format, e.g. YYYY-mm-ddTHH:MM:SS; see https://en.wikipedia.org/wiki/ISO_8601)") @RequestParam("dateNeeded") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime dateNeeded)
+            throws JsonProcessingException {
+
```

Comment:
It looks like you're missing parameters for the `done` field. Also, mutation coverage is not 100% because of this.

Reply:
Good catch, just fixed it with most recent commit